src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java

```diff
@@ -426,7 +426,19 @@ public CatalogObjectDependencies getDependencies(
                                                                                                pageSize);
 
             String accessType = "";
-            if (!grants.isEmpty()) {
+            boolean isBucketGrantExist = true;
```

Comment:
the enclosing method is too big. Could you refactor the code a bit and add private submethods with understandable names ?